api/views/session_views.py

```diff
@@ -289,37 +289,30 @@ class LectureSummaryView(APIView):
         description="Creates a summary for a specific recording using the recording ID.",
         request=LectureSummarySerializer,
         responses={
-            201: OpenApiTypes.OBJECT,  # Successful creation response
+            201: LectureSummarySerializer,  # Successful creation response
             400: OpenApiTypes.OBJECT,  # Error response for bad request
             404: OpenApiTypes.OBJECT,  # Error response when recording is not found
             500: OpenApiTypes.OBJECT,  # Error response for server error
         },
     )
     def post(self, request, recording_id):
-        summary = request.data.get("summary")
+        serializer = LectureSummarySerializer(data=request.data)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
-        # Validate required fields
-        if not summary or not recording_id:
-            return Response(
-                {"error": "Summary and recording_id are required fields."},
-                status=status.HTTP_400_BAD_REQUEST,
-            )
```

Comment:
No need, summary is validated by serializer, and recording_id is validated by Django's dispatcher.